File: services/ruby/models/post.rb
module Models
  class Post
    attr_accessor :id, :user_id, :title, :content, :published, :created_at, :updated_at

    def initialize(attributes = {})
      @id = attributes[:id]
      @user_id = attributes[:user_id]
      @title = attributes[:title]
      @content = attributes[:content]
      @published = attributes[:published] || false
      @created_at = attributes[:created_at] || Time.now
      @updated_at = attributes[:updated_at] || Time.now
    end

    def publish!
      @published = true
      @updated_at = Time.now
    end

    def unpublish!
      @published = false
      @updated_at = Time.now
    end

    def published?
      @published
    end

    def to_h
      {
        id: @id,
        user_id: @user_id,
        title: @title,
        content: @content,
        published: @published,
        created_at: @created_at,
        updated_at: @updated_at
      }
    end

    def to_json(*args)
      to_h.to_json(*args)
    end
  end
end

File: services/go/models/post.go
package models

import "time"

type Post struct {
	ID        int       `json:"id"`
	UserID    int       `json:"user_id"`
	Title     string    `json:"title"`
	Content   string    `json:"content"`
	Published bool      `json:"published"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func NewPost(userID int, title, content string) *Post {
	return &Post{
		UserID:    userID,
		Title:     title,
		Content:   content,
		Published: false,
		CreatedAt: time.Now(),
		UpdatedAt: time.Now(),
	}
}

func (p *Post) Publish() {
	p.Published = true
	p.UpdatedAt = time.Now()
}

func (p *Post) Unpublish() {
	p.Published = false
	p.UpdatedAt = time.Now()
}

func (p *Post) Update(title, content string) {
	if title != "" {
		p.Title = title
	}
	if content != "" {
		p.Content = content
	}
	p.UpdatedAt = time.Now()
}

func (p *Post) ToMap() map[string]interface{} {
	return map[string]interface{}{
		"id":         p.ID,
		"user_id":    p.UserID,
		"title":      p.Title,
		"content":    p.Content,
		"published":  p.Published,
		"created_at": p.CreatedAt,
		"updated_at": p.UpdatedAt,
	}
}
